src/traces/sankey/base_plot.js

```diff
@@ -34,3 +40,94 @@ exports.clean = function(newFullData, newFullLayout, oldFullData, oldFullLayout)
         oldFullLayout._paperdiv.selectAll('.sankey').remove();
     }
 };
+
+exports.updateFx = function(gd) {
+    for(var i = 0; i < gd._fullData.length; i++) {
+        subplotUpdateFx(gd, i);
+    }
+};
+
+var dragOptions = [];
+function subplotUpdateFx(gd, index) {
+    var i = index;
+    var fullData = gd._fullData[i];
+    var fullLayout = gd._fullLayout;
+
+    var dragMode = fullLayout.dragmode;
+    var cursor = fullLayout.dragmode === 'pan' ? 'move' : 'crosshair';
+    var bgRect = fullData._bgRect;
+
+    setCursor(bgRect, cursor);
+
+    var xaxis = {
+        _id: 'x',
+        c2p: function(v) { return v; },
+        _offset: fullData._sankey.translateX,
+        _length: fullData._sankey.width
+    };
+    var yaxis = {
+        _id: 'y',
+        c2p: function(v) { return v; },
+        _offset: fullData._sankey.translateY,
+        _length: fullData._sankey.height
+    };
+
+    // Note: dragOptions is needed to be declared for all dragmodes because
+    // it's the object that holds persistent selection state.
+    dragOptions[i] = {
+        gd: gd,
+        element: bgRect.node(),
+        plotinfo: {
+            id: i,
+            xaxis: xaxis,
+            yaxis: yaxis,
+            fillRangeItems: Lib.noop
+        },
+        subplot: i,
+        // create mock x/y axes for hover routine
+        xaxes: [xaxis],
+        yaxes: [yaxis],
+        doneFnCompleted: function(selection) {
+            var newGroups;
+            var oldGroups = gd._fullData[i].node.groups.slice();
+            var newGroup = [];
+
+            function findNode(pt) {
+                return gd._fullData[i]._sankey.graph.nodes.find(function(n) {
+                    return n.pointNumber === pt;
+                });
+            }
+
+            for(var j = 0; j < selection.length; j++) {
+                var node = findNode(selection[j].pointNumber);
+                if(!node) continue;
+
+                // If the node represents a group
+                if(node.group) {
+                    // Add all its children to the current selection
+                    for(var k = 0; k < node.childrenNodes.length; k++) {
+                        newGroup.push(node.childrenNodes[k].pointNumber);
+                    }
+                    // Flag group for removal from existing list of groups
+                    oldGroups[node.pointNumber - fullData.node._count] = false;
+                } else {
+                    newGroup.push(node.pointNumber);
+                }
+            }
+
+            newGroups = oldGroups
+                .filter(function(g) { return g; })
+                .concat([newGroup]);
+
+            Registry.call('_guiRestyle', gd, {
+                'node.groups': [ newGroups ]
+            }, i).catch(Lib.noop); // TODO will this ever fail?
+        }
+    };
+
+    dragOptions[i].prepFn = function(e, startX, startY) {
+        prepSelect(e, startX, startY, dragOptions[i], dragMode);
```

Comment:
... and I guess `prepSelect` does the right thing when `dragmode` isn't `select` nor `lasso`?

Reply:
Ahh. This is tricky actually for `sankey`, dragmode values `zoom` and `pan` don't really make sense. I'm not sure if we can do anything about it in v1. Would it be ok to have `zoom` and `pan` correspond to the current "drag-on-nodes" behaviour?

Reply:
> Would it be ok to have zoom and pan correspond to the current "drag-on-nodes" behavior?

Yes, I think it would be OK. I will return early in `subplotUpdateFx` when `dragMode` is either `zoom` or `pan`.

Reply:
Done in aa17f1b6b32091fc171e97dfad4e3e18ee331421